webapp/src/app/app.component.ts

```diff
@@ -50,12 +51,21 @@ export class AppComponent implements OnInit {
                         window.location.reload();
                     } else {
                         this.loading = false;
-                        await this.router.navigate(["/newuser"]);
+                        return this.newUser();
                     }
                 });
             }
         }
     }
+
+    async newUser() {
```

Comment:
`private`, what's the return type (it's boolean but it is not used/relevant so maybe tagging it at `Promise<void>`)?